src/util/strb.c

```diff
@@ -34,24 +34,35 @@ int strb_grow(strb *sb, size_t n) {
   return 0;
 }
 
-void strb_appendf(strb *sb, const char *f, ...) {
-  va_list ap;
-  int s;
+void strb_appendv(strb *sb, const char *f, va_list ap) {
+  va_list apSave;
+  int     s;
 
-  va_start(ap, f);
 #ifdef _MSC_VER
-  s = _vscprintf(f, ap);
+  /**
+   * va_copy() is a C99 novelty that a particular company should have started
+   * supporting a long time ago, to their undying shame.
+   */
+  
+  apSave = ap;
+  s = _vscprintf(f, apSave);
 #else
-  s = vsnprintf(NULL, 0, f, ap);
+  va_copy(apSave, ap);
+  s = vsnprintf(NULL, 0, f, apSave);
 #endif
-  va_end(ap);
-
+  va_end(apSave);
```

Comment:
The `va_end()` is in the correct place.
- `va_copy()`ing a `va_list` is equivalent to `va_start()`ing it followed by an equivalent number of `va_arg()` calls.
- Every `va_start()` must be matched by a `va_end()`.
- `va_copy()` doesn't exist on older Visual Studios, but in the Visual Studios where it's implemented it's implemented as just a straight assignment.

Therefore `va_end()` should be unconditional or else it's UB (although on x86-64 it's actually a no-op)